app/models/grade_entry_form.rb

```diff
@@ -131,16 +137,25 @@ def export_as_csv(role)
                        .group_by { |x| x[0] }
       num_items = self.grade_entry_items.count
     end
-    MarkusCsv.generate(students, headers) do |user_name, total_grade|
-      row = [user_name]
-      if grade_data.key? user_name
+
+    MarkusCsv.generate(students, headers) do |student|
+      row = Student::CSV_ORDER.map do |field|
+        student[field]
+      end
+      if grade_data.key? student[:user_name]
         student_grades = Array.new(num_items, '')
-        grade_data[user_name].each do |g|
+        grade_data[student[:user_name]].each do |g|
           grade_index = g[1] - 1
           student_grades[grade_index] = g[2].nil? ? '' : g[2]
         end
         row.concat(student_grades)
-        row << (total_grade.nil? ? '' : total_grade) if self.show_total
+        if self.show_total
```

Comment:
This is a bit more awkward than the original. Better to define a ` total_grade` variable

Reply:
Look carefully at the git diff here. The reason I said to define a variable was so that this change could be reverted.